pkg/sql/randgen/mutator.go

```diff
@@ -742,10 +742,21 @@ func postgresCreateTableMutator(
 			for _, def := range stmt.Defs {
 				switch def := def.(type) {
 				case *tree.ColumnTableDef:
-					colTypes[string(def.Name)] = tree.MustBeStaticallyKnownType(def.Type)
+					colType := tree.MustBeStaticallyKnownType(def.Type)
+					// Disable locales because CRDB random generator generates
+					// locales without double-quotes, and they are unknown to
+					// Postgres (e.g. `i TEXT COLLATE de_DE` won't work with
```

Comment:
another alternative is to always add double quotes here, i think:
```
// Format implements the NodeFormatter interface.
func (node *CollateExpr) Format(ctx *FmtCtx) {
	exprFmtWithParen(ctx, node.Expr)
	ctx.WriteString(" COLLATE ")
	lex.EncodeLocaleName(&ctx.Buffer, node.Locale)
}
```

Reply:
It will break a lot of existing tests, and also need to replace "_" with "-" to make it work. It's too risky and complicated so let's just disable it.